keel-core-test/src/main/kotlin/com/netflix/spinnaker/keel/persistence/CleanerTests.kt

```diff
@@ -0,0 +1,158 @@
+package com.netflix.spinnaker.keel.persistence
+
+import com.netflix.spinnaker.keel.api.ApiVersion
+import com.netflix.spinnaker.keel.api.DeliveryConfig
+import com.netflix.spinnaker.keel.api.DockerArtifact
+import com.netflix.spinnaker.keel.api.Environment
+import com.netflix.spinnaker.keel.api.ResourceSpec
+import com.netflix.spinnaker.keel.api.id
+import com.netflix.spinnaker.keel.api.resources
+import com.netflix.spinnaker.keel.resources.ResourceTypeIdentifier
+import com.netflix.spinnaker.keel.test.DummyResourceSpec
+import com.netflix.spinnaker.keel.test.resource
+import dev.minutest.junit.JUnit5Minutests
+import dev.minutest.rootContext
+import io.mockk.mockk
+import java.time.Clock
+import org.springframework.context.ApplicationEventPublisher
+import strikt.api.expect
+import strikt.api.expectThat
+import strikt.api.expectThrows
+import strikt.assertions.isEmpty
+import strikt.assertions.isEqualTo
+
+abstract class CleanerTests<T : DeliveryConfigRepository, R : ResourceRepository, A : ArtifactRepository> :
```

Comment:
Might as well go with a `D` here for the `DeliveryConfigRepository` type for consistency... 😄